compose/compose.go

```diff
@@ -116,6 +136,36 @@ func (c *Compose) MustKill() {
 	}
 }
 
+// GetPublicIPAddressForService returns the IPAddress of a service
+func (c *Compose) GetPublicIPAddressForService(serviceName string) (bool, string) {
```

Comment:
So for example on my machine, if I do:

```bash
$ docker run -d jamesdbloom/mockserver
$ docker ps
```

I get

```
279eff516fcb        jamesdbloom/mockserver           "/opt/mockserver/run_"   About a minute ago   Up About a minute   1080/tcp, 1090/tcp                 festive_hopper
```

Then

```
$ docker inspect festive_hopper
```

I get 

```
...
"NetworkSettings": {
            "Bridge": "",
            "SandboxID": "200a8403eeae9d7e40c68a7db142cf5a24a103c24e7ad5690bb64ca79b5cd271",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {
                "1080/tcp": null,
                "1090/tcp": null
            },
            "SandboxKey": "/var/run/docker/netns/200a8403eeae",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "b26b58b68f82f0bbee026b4673359260fab8912f113050b4c74d70de9c784873",
            "Gateway": "172.17.0.1",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "172.17.0.4",
            "IPPrefixLen": 16,
            "IPv6Gateway": "",
            "MacAddress": "02:42:ac:11:00:04",
            "Networks": {
```

Then

```
$ curl -v http://172.17.0.4:1080
````

It times out.  But:

```
$ docker run -d -p 1080 jamesdbloom/mockserver
$ docker ps
```

Gives me 

```
d2f56406d942        jamesdbloom/mockserver   "/opt/mockserver/run_"   3 seconds ago       Up 2 seconds        1090/tcp, 0.0.0.0:32769->1080/tcp   vigilant_torvalds
```

Which tells me that I can connect to this container using port 32769 on the Docker host. In my case, since I'm using boot2docker, it would be:

```
$ curl -v http://192.168.99.100:32769
```

Which works! If I did:

```
$ docker run -d -p 1080:1080 jamesdbloom/mockserver
$ docker run -d -p 1080:1080 jamesdbloom/mockserver
```

The first one would succeed and I could connect using `curl -v http://192.168.99.100:1080`, the second one would fail with this error message:

```
docker: Error response from daemon: driver failed programming external connectivity on endpoint relaxed_saha (fd8eeb6e8c75baeda60698af4205595efb7765565568080b81464236a852315f): Bind for 0.0.0.0:1080 failed: port is already allocated.
```